packages/browser-pool/src/playwright/playwright-controller.ts

```diff
@@ -75,26 +75,30 @@ export class PlaywrightController extends BrowserController<BrowserType, Paramet
                     (this.browserPlugin as PlaywrightPlugin)._containerProxyServer!.ipToProxy.set(proxyip, url.href);
                 }
 
-                const session = await page.context().newCDPSession(page);
-                await session.send('Network.enable');
-
-                session.on('Network.responseReceived', (responseRecevied) => {
-                    const logOnly = ['Document', 'XHR', 'Fetch', 'EventSource', 'WebSocket', 'Other'];
-                    if (!logOnly.includes(responseRecevied.type)) {
-                        return;
-                    }
-
-                    const { response } = responseRecevied;
-                    if (response.fromDiskCache || response.fromPrefetchCache || response.fromServiceWorker) {
-                        return;
-                    }
-
-                    const { remoteIPAddress } = response;
-                    if (remoteIPAddress && remoteIPAddress !== proxyip) {
-                        // eslint-disable-next-line no-console
-                        console.warn(`Request to ${response.url} was through ${remoteIPAddress} instead of ${proxyip}`);
-                    }
-                });
+                try {
+                    const session = await page.context().newCDPSession(page);
+                    await session.send('Network.enable');
+
+                    session.on('Network.responseReceived', (responseRecevied) => {
+                        const logOnly = ['Document', 'XHR', 'Fetch', 'EventSource', 'WebSocket', 'Other'];
+                        if (!logOnly.includes(responseRecevied.type)) {
+                            return;
+                        }
+
+                        const { response } = responseRecevied;
+                        if (response.fromDiskCache || response.fromPrefetchCache || response.fromServiceWorker) {
+                            return;
+                        }
+
+                        const { remoteIPAddress } = response;
+                        if (remoteIPAddress && remoteIPAddress !== proxyip) {
+                            // eslint-disable-next-line no-console
+                            console.warn(`Request to ${response.url} was through ${remoteIPAddress} instead of ${proxyip}`);
+                        }
+                    });
+                } catch {
+                    // Firefox does not support this
```

Comment:
this will silence all errors in general, maybe better to check if we are using firefox explicitly? or at least log it?

Reply:
There are only two possible solutions here. Either it throws because it's Firefox or we got disconnected from the browser. I'll add `if (notFirefox) throw error` for readability.

Reply:
Or even better, wrap this in an `if (notFirefox) { code }`